haystack_experimental/components/embedders/sentence_transformers_doc_image_embedder.py

```diff
@@ -0,0 +1,335 @@
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        serialization_dict = default_to_dict(
+            self,
+            file_path_meta_field=self.file_path_meta_field,
+            root_path=self.root_path,
+            size=self.size,
+            model=self.model,
+            device=self.device.to_dict(),
+            token=self.token.to_dict() if self.token else None,
+            batch_size=self.batch_size,
+            progress_bar=self.progress_bar,
+            trust_remote_code=self.trust_remote_code,
+            local_files_only=self.local_files_only,
+            model_kwargs=self.model_kwargs,
+            tokenizer_kwargs=self.tokenizer_kwargs,
+            config_kwargs=self.config_kwargs,
+            precision=self.precision,
+            encode_kwargs=self.encode_kwargs,
+            backend=self.backend,
+        )
+        if serialization_dict["init_parameters"].get("model_kwargs") is not None:
+            serialize_hf_model_kwargs(serialization_dict["init_parameters"]["model_kwargs"])
+        return serialization_dict
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "SentenceTransformersDocumentImageEmbedder":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            Dictionary to deserialize from.
+        :returns:
+            Deserialized component.
+        """
+        init_params = data["init_parameters"]
+        if init_params.get("device") is not None:
+            init_params["device"] = ComponentDevice.from_dict(init_params["device"])
+        deserialize_secrets_inplace(init_params, keys=["token"])
+        if init_params.get("model_kwargs") is not None:
+            deserialize_hf_model_kwargs(init_params["model_kwargs"])
+        return default_from_dict(cls, data)
+
+    def warm_up(self) -> None:
+        """
+        Initializes the component.
+        """
+        if self._embedding_backend is None:
+            self._embedding_backend = _SentenceTransformersEmbeddingBackendFactory.get_embedding_backend(
+                model=self.model,
+                device=self.device.to_torch_str(),
+                auth_token=self.token,
+                trust_remote_code=self.trust_remote_code,
+                local_files_only=self.local_files_only,
+                model_kwargs=self.model_kwargs,
+                tokenizer_kwargs=self.tokenizer_kwargs,
+                config_kwargs=self.config_kwargs,
+                backend=self.backend,
+            )
+            if self.tokenizer_kwargs and self.tokenizer_kwargs.get("model_max_length"):
+                self._embedding_backend.model.max_seq_length = self.tokenizer_kwargs["model_max_length"]
+
+    def _validate_image_paths(self, documents: List[Document]) -> List[Dict[str, str]]:
```

Comment:
Do you think it's worth making this a utility function so we could reuse it as a utility in the `DocumentToImageContent` component?

Reply:
It makes sense, but I would do this in a subsequent PR to keep the current one relatively focused.

Reply:
sounds good!